### ISSUE TITLE
"Other applications" under "Focus/select tabs that open from" is broken in Fx110

### ISSUE THREAD

**Issue by purple512** [onemen/TabMixPlus#209]
I like to have tabs opened from external applications not focused. This worked perfectly by unchecking the box in the dialog.

![tmp-other-applications](https://user-images.githubusercontent.com/79816784/219530578-429d07f7-8def-4a62-9068-b012b999bfb4.png)

This stopped working in Firefox 110. The tab now opens in the foreground.

**Comment by purple512** [comment]
Oh, I see that unchecking "Diverted windows" now seems to prevent tabs opened from external applications from being focused, so I guess that's a workaround. But this was never necessary before so it's still something different about Fx110.

**Comment by onemen** [comment]
New version coming soon....

**Comment by onemen** [comment]
Download latest version [v1.0.0-pre.13](https://github.com/onemen/TabMixPlus/releases)

**Comment by purple512** [comment]
I installed pre.13 but it has the same behavior, that is, the setting of "Diverted windows" now determines whether links opened from other applications receive focus, and the setting of "Other applications" seems to have no effect.

I'm OK with just using the Diverted windows setting but I'm not entirely sure what the distinction between the two is meant to entail. I haven't noticed any unwanted behavior of deselecting Diverted windows but I've only been using that setting for a short time.

**Comment by onemen** [comment]
Can you provide steps to reproduce/test this issue ?

**Comment by purple512** [comment]
I have Firefox as my default browser. If I do some action in another program that wants to open a URL (e.g. in Sublime Text, click on Help -> Documentation which opens a website) either Firefox gets focus and is brought to the front or the invoking program stays in focus and the new tab is opened in the background and Firefox's window is not activated (desired behavior.)

In pre.13:

Diverted windows checked, Other applications unchecked -> new tab gets focus (change from Fx109)
Diverted windows unchecked, Other applications unchecked -> new tab not focused
Diverted windows unchecked, Other applications checked -> new tab not focused

So only "Diverted windows" seems to matter.

I'll attach my config file, I don't think there's any sensitive information contained within.

[TMPpref-2023-02-17.txt](https://github.com/onemen/TabMixPlus/files/10765606/TMPpref-2023-02-17.txt)

**Comment by purple512** [comment]
I should clarify that I have no idea what a Diverted window even is. I guess that's part of my question. All I care about is what happens to tabs opened from other programs, which prior to Fx110 was controlled by "Other programs" setting.

**Comment by purple512** [comment]
Clearing the startup cache didn't have any effect.

I will try creating a new browser profile with just TMP and userchromejs to see if I can reproduce it with as few changes from stock as possible.

**Comment by purple512** [comment]
This is very bizarre. I created a brand new profile, installed userChromeJS and TabMixPlus, couldn't reproduce the issue.

Part of testing required making the new profile the default, so I set the default profile setting back to my main profile to start testing it for extension compatibilities. When I did that, it started working again. Something about switching the default profile and then back fixed it. I cannot fathom how or why that would be the case, but it seems to have done it. Marking as closed.
